model/src/main/java/com/javahelp/model/token/Token.java

```diff
@@ -0,0 +1,123 @@
+package com.javahelp.model.token;
+
+import com.javahelp.model.user.User;
+import org.joda.time.DateTime;
+import java.security.SecureRandom;
+
+/**
+ * Representation of token entity
+ */
+
+public class Token {
+    /**
+     * Token requested by the user
+     */
+    private String token;
+
+    /**
+     * Unique id of the token.
+     */
+    private String id;
+
+    /**
+     * Expiry date of the token.
+     */
+    private DateTime expiry;
+
+    /**
+     * Issued date of the token.
+     */
+    private DateTime issued;
+
+    /**
+     * User who requested the token.
+     */
+    private User user;
+
+    /**
+     * This holds any additional information about the toekn.
+     */
+    private String tag;
+
+
+    /**
+     * Creates a new {@link Token}
+     * @param tag {@link String} of this {@link Token}
+     * @param user {@link User} who requested this {@link Token}
+     */
+    public Token(String tag, User user){
+        // Creating a token using functions from SecureRandom
+
+        SecureRandom random = new SecureRandom();
+        byte bytes[] = new byte[20];
+        random.nextBytes(bytes);
+        this.token = bytes.toString();
+
+        this.issued = new DateTime(DateTime.now());
+        this.expiry = this.issued.plusDays(365/2);
```

Comment:
We need the option to set both these fields from the constructor. We may want some tokens to expire sooner or later than others (i.e. User is signed on permanently, vs signed on for single session). We will also want to be able to set the issued date to any arbitrary date, as a `Token` object may not always be created on the date it was issued. Consider a token issued 2 weeks ago, but used to login in the present. When the backend authenticates this token, it will create a `Token` object, and will need to create one with a issued date of two weeks prior.

Reply:
Great catch, one question though, should the Database of Token be responsible for telling when was that token created before and how long more it will be valid (if true)?

Then we might need these parameters from the Database, I guess.

Reply:
Yeah, that's exactly right